--- origsrc/pulseaudio-1.1/src/Makefile.am	2011-10-20 07:54:16.000000000 -0500
+++ src/pulseaudio-1.1/src/Makefile.am	2011-11-22 03:20:45.254582500 -0600
@@ -25,7 +25,7 @@
 
 pulseincludedir=$(includedir)/pulse
 pulsecoreincludedir=$(includedir)/pulsecore
-pulselibexecdir=$(libexecdir)/pulse
+pulselibexecdir=$(libexecdir)/pulse-@PA_MAJORMINOR@
 if HAVE_X11
 xdgautostartdir=$(sysconfdir)/xdg/autostart
 endif
