--- origsrc/pulseaudio-1.1/configure.ac	2011-10-20 08:12:05.000000000 -0500
+++ src/pulseaudio-1.1/configure.ac	2011-11-21 17:50:02.605015000 -0600
@@ -1108,7 +1108,7 @@ AX_DEFINE_DIR(PA_BINARY, PA_BINARY, [Loc
 PACTL_BINARY=${bindir}/pactl${EXEEXT}
 AX_DEFINE_DIR(PACTL_BINARY, PACTL_BINARY, [Location of pactl binary])
 
-AC_SUBST(PA_SOEXT, [.so])
+AC_SUBST(PA_SOEXT, [$shrext_cmds])
 
 AC_SUBST(pulseconfdir, ["${sysconfdir}/pulse"])
 AX_DEFINE_DIR(PA_DEFAULT_CONFIG_DIR, pulseconfdir, [Location of configuration files])
