--- origsrc/pulseaudio-1.1/src/Makefile.am	2011-10-20 07:54:16.000000000 -0500
+++ src/pulseaudio-1.1/src/Makefile.am	2011-11-22 03:20:45.254582500 -0600
@@ -52,14 +52,14 @@ SERVER_CFLAGS = -D__INCLUDED_FROM_PULSE_
 
 AM_LIBADD = $(PTHREAD_LIBS) $(INTLLIBS)
 AM_LDADD = $(PTHREAD_LIBS) $(INTLLIBS)
-AM_LDFLAGS = $(NODELETE_LDFLAGS)
+AM_LDFLAGS = $(NODELETE_LDFLAGS) -no-undefined
 
 if STATIC_BINS
 BINLDFLAGS = -static
 endif
 
 if OS_IS_WIN32
-AM_LDFLAGS+=-Wl,--export-all-symbols,--enable-auto-import -no-undefined
+AM_LDFLAGS+=-Wl,--export-all-symbols,--enable-auto-import
 WINSOCK_LIBS=-lwsock32 -lws2_32 -lwininet
 endif
 
@@ -1838,18 +1838,10 @@ if HAVE_SPEEX
 module_echo_cancel_la_SOURCES += modules/echo-cancel/speex.c
 endif
 if HAVE_WEBRTC
-# The webrtc code is split off into a helper library to avoid having automake
-# link module-echo-cancel with C++ (which it does if there are any C++ deps,
-# even conditional ones).
-modlibexec_LTLIBRARIES += libwebrtc-util.la
-
-libwebrtc_util_la_SOURCES = modules/echo-cancel/webrtc.cc
-libwebrtc_util_la_CXXFLAGS = $(AM_CXXFLAGS) $(SERVER_CFLAGS) $(WEBRTC_CFLAGS) -DHAVE_WEBRTC=1
-libwebrtc_util_la_LIBADD = $(WEBRTC_LIBS)
-libwebrtc_util_la_LDFLAGS = -avoid-version
-
+module_echo_cancel_la_SOURCES += modules/echo-cancel/webrtc.cc
 module_echo_cancel_la_CFLAGS += -DHAVE_WEBRTC=1
-module_echo_cancel_la_LIBADD += libwebrtc-util.la
+module_echo_cancel_la_CXXFLAGS = $(AM_CXXFLAGS) $(SERVER_CFLAGS) $(WEBRTC_CFLAGS) -DHAVE_WEBRTC=1
+module_echo_cancel_la_LIBADD += $(WEBRTC_LIBS) libpulsecore-@PA_MAJORMINOR@.la libpulsecommon-@PA_MAJORMINOR@.la libpulse.la
 endif
 
 # RTP modules
