Requires SO_TIMESTAMP

--- origsrc/pulseaudio-1.1/src/Makefile.am	2011-10-20 07:54:16.000000000 -0500
+++ src/pulseaudio-1.1/src/Makefile.am	2011-11-22 03:20:45.254582500 -0600
@@ -892,7 +892,7 @@ modlibexec_LTLIBRARIES = \
 		libprotocol-esound.la
 
 # We need to emulate sendmsg/recvmsg to support this on Win32
-if !OS_IS_WIN32
+if !HAVE_WAVEOUT
 modlibexec_LTLIBRARIES += \
 		librtp.la
 endif
@@ -1007,7 +1007,7 @@ modlibexec_LTLIBRARIES += \
 		module-filter-heuristics.la
 
 # See comment at librtp.la above
-if !OS_IS_WIN32
+if !HAVE_WAVEOUT
 modlibexec_LTLIBRARIES += \
 		module-rtp-send.la \
 		module-rtp-recv.la
@@ -1196,6 +1196,7 @@ pulselibexec_PROGRAMS += \
 		proximity-helper
 endif
 
+if !HAVE_WAVEOUT
 if HAVE_OPENSSL
 modlibexec_LTLIBRARIES += \
 		libraop.la \
@@ -1205,6 +1205,7 @@ modlibexec_LTLIBRARIES += \
 		module-raop-discover.la
 endif
 endif
+endif
 
 if HAVE_DBUS
 if HAVE_FFTW
